src/qt/walletmodel.cpp

```diff
@@ -756,3 +774,258 @@ bool WalletModel::getDefaultWalletRbf() const
+        // Make sure we have the correct error response and not something else
+        UniValue message = find_value( e, "message");
+        std::string errorStr = message.get_str();
+
+        if (errorStr.find("name not found") != std::string::npos)
+            return true;
+
+        if(reason)
+            *reason = QString::fromStdString(errorStr);
+
+        LogPrintf ("unexpected nameAvailable response: %s\n", errorStr.c_str());
+        return false;
+    }
+
+    isExpired = find_value( res, "expired");
+    if (isExpired.get_bool())
+      return true;
+
+    return false;
+}
+
+WalletModel::NameNewReturn WalletModel::nameNew(const QString &name)
+{
+    std::string strName = name.toStdString ();
+    std::string data;
+
+    std::vector<UniValue> values;
+
+    UniValue params(UniValue::VOBJ);
+    params.pushKV("name", strName);
+
+    JSONRPCRequest jsonRequest;
+    jsonRequest.strMethod = "name_new";
+    jsonRequest.params = params;
+    jsonRequest.fHelp = false;
+
+    NameNewReturn retval;
+    UniValue res;
+    try {
+        res = tableRPC.execute(jsonRequest);
+    } catch (const UniValue& e) {
+        UniValue message = find_value( e, "message");
+        std::string errorStr = message.get_str();
+        LogPrintf ("nameNew error: %s\n", errorStr.c_str());
+        retval.err_msg = errorStr;
+        retval.ok = false;
+        return retval;
+    }
+
+    values = res.getValues();
+    const auto txid = values[0];
+    const auto  rand = values[1];
+
+    retval.ok = true;
+    retval.hex = txid.get_str(); // txid
+    retval.rand = rand.get_str();
+
+    return retval;
+}
+
+const std::string WalletModel::completePendingNameFirstUpdate(const std::string &name, const std::string &rand, const std::string &txid, const std::string &data, const std::string &toaddress)
+{
+    std::string errorStr;
+    LogPrintf ("WalletModel::completePendingNameFirstUpdate: %s\n", name.c_str());
+
+    UniValue params(UniValue::VOBJ);
+    params.pushKV ("name", name);
+    params.pushKV ("rand", rand);
+    params.pushKV ("tx", txid);
+    params.pushKV ("value", data);
+    if (!toaddress.empty())
+        params.pushKV ("toaddress", toaddress);
+
+    JSONRPCRequest jsonRequest;
+    jsonRequest.strMethod = "name_firstupdate";
+    jsonRequest.params = params;
+    jsonRequest.fHelp = false;
+
+    LogPrintf("executing name_firstupdate name=%s rand=%s tx=%s value=%s\n",
+        name.c_str(), rand.c_str(), txid.c_str(), data.c_str());
+
+    UniValue res;
+    try {
+        res = tableRPC.execute(jsonRequest);
+    }
+    catch (const UniValue& e) {
+        UniValue message = find_value( e, "message");
+        errorStr = message.get_str();
+        LogPrintf ("name_firstupdate error: %s\n", errorStr.c_str());
+    }
+
+    return errorStr;
+}
+
+std::vector<std::reference_wrapper<std::string>> WalletModel::sendPendingNameFirstUpdates()
+{
+    std::vector<std::reference_wrapper<std::string>> successfulNames;
```

Comment:
Do you really need a `reference_wrapper<string>` here?  I think just having a `vector<string>` would make the code simpler and easier to read, besides avoiding any potential issues with references to strings that go out of scope.  Of course, a vector of references is smaller and faster, but is that really important here vs code simplicity and clarity?